tsdb/head.go

```diff
@@ -1915,7 +1920,8 @@ type memSeries struct {
 
 	mmMaxTime int64 // Max time of any mmapped chunk, only used during WAL replay.
 
-	nextAt int64 // Timestamp at which to cut the next chunk.
+	samplesPerChunk int   // Target number of samples per chunk.
```

Comment:
We should probably make this `uint16` and have validation during init to keep it within that range. Saves 48 bytes per series.

Reply:
We ended up pulling that out in https://github.com/prometheus/prometheus/pull/12390.